### PR TITLE
Improve CI

### PR DESCRIPTION
**Pull request by szepeviktor**
kép

Voila! :)

### CODE REVIEW

File: .editorconfig [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+[**/*.neon]
+indent_style = space
+indent_size = 4
+
+[**/*.js]
+indent_style = space
+indent_size = 2
+
+[composer.lock]
+indent_style = unset
```
Comment (szepeviktor) [inline .editorconfig:42]: Please consider using only spaces (or only tabs).

File: .github/workflows/integrate.yml [hunk 2]
```diff
@@ -0,0 +1,192 @@
...
+        php-version:
+          - "8.2"
+          - "8.1"
+          - "8.0"
+          - "7.4"
+          - "7.3"
+        dependencies:
+          - "lowest"
+          - "locked"
+          - "highest"
```
Comment (szepeviktor) [inline .github/workflows/integrate.yml:84]: `locked` could be only used with your local PHP version.

File: composer.json [hunk 3]
```diff
@@ -28,14 +28,16 @@
   },
   "require-dev": {
     "brain/monkey": "^2.6",
+    "mockery/mockery": "^1.0",
```
Comment (szepeviktor) [inline composer.json:31]: Unit tests still fail with `lowest` deps.

File: tests/unit/src/Cases/XmlSitemapReader/XmlSitemapTest.php [hunk 4]
```diff
@@ -1,6 +1,6 @@
 <?php
 
-namespace BlueChip\Cache\Tests\Unit\Cases\Core;
+namespace BlueChip\Cache\Tests\Unit\Cases\XmlSitemapReader;
```
Comment (szepeviktor) [inline tests/unit/src/Cases/XmlSitemapReader/XmlSitemapTest.php:3]: # 💣

File: .github/workflows/integrate.yml [hunk 5]
```diff
@@ -0,0 +1,192 @@
...
+        uses: "actions/checkout@v3"
+
+      - name: "Check file permissions"
+        run: |
+          test "$(find . -type f -not -path './.git/*' -executable)" = ""
+
+      - name: "Find non-printable ASCII characters"
+        run: |
+          #! LC_ALL=C.UTF-8 find . -type f -name '*.php' -print0 | xargs -0 -- grep -PHn '[^ -~Č]' | grep -v '// @ignore-non-ascii$'
+          ! LC_ALL=C.UTF-8 find . -type f -name '*.php' -print0 | xargs -0 -- grep -PHn '[^ -~Č]'
```
Comment (szepeviktor) [inline .github/workflows/integrate.yml:43]: 🇨🇿 There is only a single `Č` character in the whole code base :)

**Comment by chesio** [comment]
Thank you Viktor! I'll take some time to review - plenty of new stuff for me 😮

**Comment by szepeviktor** [comment]
This lowest deps thing is tough stuff! I'm not jealous of you.

**Comment by chesio** [comment]
> This lowest deps thing is tough stuff! I'm not jealous of you.

Sadly I have no idea how to deal with reported the error - there is nothing returned from the mentioned method, it's as void as it can be... 🆘

File: .editorconfig (hunk 1 above)
Comment (chesio) [reply .editorconfig]: I like to have the config files (.json, .yaml...) more compact, so I'll keep them with 2 spaces only. Markdown as well. With anything else I prefer spaces over tabs.

**Comment by chesio** [comment]
@szepeviktor Problem solved! 😄 

Just for fun I have tracked it down to this issue: https://github.com/antecedent/patchwork/issues/64 So I would have to add Patchwork as direct dependency just to pass the `lowest` tests, not really worth it.

**Comment by chesio** [comment]
Thanks a lot again! And now back to work... 💪

**Comment by szepeviktor** [comment]
You're welcome.

**Pull request merged**